src/Generator.php

```diff
@@ -56,8 +56,9 @@ public function generate(string $className, array $context = []) : array
             $propertyInfoContext = $context['property_info_context'] ?? [];
 
             $writable = $this->propertyInfoExtractor->isWritable($className, $propertyName, $propertyInfoContext);
+            $readable = $this->propertyInfoExtractor->isReadable($className, $propertyName, $propertyInfoContext);
 
-            if (null === $writable || false === $writable) {
+            if ((null === $writable || false === $writable) && null === $readable || false === $readable) {
```

Comment:
Can you add parenthesis on the right expression for clarity?

Reply:
Done.